### DISCUSSION TITLE
Why are enums restricted as to the member identifiers?

### DISCUSSION THREAD

**Discussion by pbelbin** [prisma/prisma#4952] [Q&A] [answered]
Enums are currently being limited from having '.' and other non-alphanumeric values, and also prevent it starting with a numeric.

Currently, on the JavaScript/TypeScript side of things, the enums appear to be organized and exposed as arbitrary objects, and in this case, there really is no requirement to limit the content, as, each key is really a string, and not limited as to what it can contain.

So, what is the reason for limiting the enum so that '.' and other text values are not permitted?

eg:

Prisma schema could allow:

```prisma
enum ThingTypes {
   One
   "3.4"
   "4 and 5"
}
```

Instead, we are required to avoid numbers and periods and such, and use `@map()` instead:

```prisma
enum ThingTypes {
   One
   ThreeDotFour @map("3.4")
   FourAndFive @map("4 and 5")
}
```

So, why is this?

**Comment by nikolasburk** [comment, marked as answer]
Hey Peter, as we discussed on Slack there doesn't seem to be an underlying technical reason for this limitation in enum values and it's likely to be resolved soon (you can follow the progress in the [issue](https://github.com/prisma/prisma/issues/4954) you created). Thanks for raising this 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
